src/apps/opening-hours-editor/EventForm.tsx

```diff
@@ -4,124 +4,246 @@
+          data-cy="opening-hours-editor-form-start-time"
+          className="opening-hours-editor-form__time-input"
+          id="event-form-start-time"
+          type="time"
+          value={startTime}
+          onChange={(e) => setStartTime(e.target.value)}
+        />
+        <label
+          className="opening-hours-editor-form__label"
+          htmlFor="event-form-end-time"
+        >
+          {t("openingHoursEventFormEndTimeText")}
+        </label>
+        <input
+          data-cy="opening-hours-editor-form-end-time"
+          className="opening-hours-editor-form__time-input"
+          id="event-form-end-time"
+          type="time"
+          value={endTime}
+          onChange={(e) => setEndTime(e.target.value)}
+          min={startTime}
+          max="00:00"
+        />
+        {isRepeatedOpeningHour && (
+          <>
+            <div className="opening-hours-editor-form__checkbox">
+              <input
+                id="event-form-repeated"
+                data-cy="opening-hours-editor-form-repeated"
+                type="checkbox"
+                checked={isRepeated}
+                onChange={(e) => setIsRepeated(e.target.checked)}
+              />
+              <label
+                className="opening-hours-editor-form__label"
+                htmlFor="event-form-repeated"
+              >
+                {t("openingHoursEventFormRepeatedText", {
+                  placeholders: { "@startDate": startDateString }
+                })}
+              </label>
+            </div>
+
+            <div className="opening-hours-editor-form__checkbox">
+              <input
+                id="event-form-weekly"
+                data-cy="opening-hours-editor-form-week-day"
+                type="checkbox"
+                checked={isRepeated}
+                disabled={!isRepeated}
+              />
+              <label className="" htmlFor="event-form-weekly">
+                {t("openingHoursEventFormWeklyText")}
+              </label>
+            </div>
+
+            <div className="opening-hours-editor-form__checkbox">
+              <input
+                id="event-form-week-day"
+                data-cy="opening-hours-editor-form-week-day"
+                type="checkbox"
+                checked={isRepeated}
+                disabled={!isRepeated}
+              />
+              <label className="" htmlFor="event-form-week-day">
+                {weekDayName}
+              </label>
+            </div>
+
+            <label
+              className="opening-hours-editor-form__label"
+              htmlFor="event-form-end-date"
+            >
+              {t("openingHoursEventFormEndDateText")}
+            </label>
+            <input
+              type="date"
+              className="opening-hours-editor-form__time-input"
+              id="event-form-end-date"
+              min={getStringForDateInput(startDate)}
+              disabled={!isRepeated}
+              required={isRepeated}
```

Comment:
:+1: for controlling state and values here. Adding the start date as a minimum value is helpful and well as remembering to set the field as required.